docs/validators/emergency-upgrades.md

```diff
@@ -0,0 +1,83 @@
+---
+order: 5
+title: Emergency Upgrades
+---
+
+## **Introduction**
+
+This article goes into the general flow of events for an emergency upgrade, pre-upgrade and during the upgrade for all validators.
+
+Emergency upgrades are an unfortunate necessity for all validators when running a PoS node.
+
+When a vunerability is discovered and responsibily disclosed, the vunerability will be reported to one of the channels described in the [Security.md](../../SECURITY.md).
+
+Once the vulnerability is patched and tested, the core team will reach out to all validators to update their nodes with the latest version.
```

Comment:
there is no score team of the Cosmos Hub. This would be one of the teams working on the software development of the Hub.

Reply:
This is definitely a sensitive topic because we're in the middle of figuring it out.

Right now, reach out is done by Milan&team who are core developers of the gaia repo. Even for Comet or SDK fixes, it's Milan&team doing the reach out for the Cosmos Hub. (Other networks do it differently.)

But it doesn't specifically have to be the Hub team doing the reach-out. I hope at one point Milan gets his own team dealing with all kinds of communications but that's just one option we can think of.

Amulet is helping in the coordination but they are also on the side that the authenticity of a message if it comes from the source: if the core team has a fix, they're the ones who should be advertising it.

So, maybe it's best not to address this yet, since it's in flux.

It's a much more important question of how to authenticate the message when one goes out. I was accused of a hacked account when I submitted a post to the forum (I'm a private person and I don't post much otherwise). But if anyone with half a brain saw the message, they could've deduced that I'm pointing to the _official_ repository that will contain the fix and not trying to derail anyone elsewhere. I think having the official repositories corroborate a message is a good way to authenticate it.

The _current_ (might change in the future) outline for communications from a security perspective is:
1. Heads-up - message on the forum. The point of it is to prepare the validators for an upcoming fix in the next day or two. It could also establish an official communications channel for coordination. (The barberry patch worked out well in the end and networks had direct access to all information available in an easily managed format.)
2. Release - this is on the official channel (GitHub) and it's the main validated information for validators. It's not a message, it's an actual release of the source code.
3. Deployment - validator coordination and one-to-one communications to help them push through. I think we can improve this in the future: even though Milan&et al are doing a great job, we need to simplify this communication. Also, the amulet team is not going to take this over.
4. Advisory - this happens when all the impacted parties are upgraded and generally comes from the core team responsible for the release. OR the team responsible for patching the source code. For example a Comet advi

Reply:
Added this content to the doc, thxs @greg-szabo !